server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -638,6 +638,43 @@ public void run() {
    */
   private Map<DataLevel,Set<KeyExtent>> partitionMigrations() {
     final Map<DataLevel,Set<KeyExtent>> partitionedMigrations = new EnumMap<>(DataLevel.class);
+
+    // Don't try to check migrations if the Root and Metadata
+    // tables are not hosted.
+    boolean skipMigrationCheck = false;
+
+    if (watchers.size() != 3) {
+      log.debug("Skipping migration check, not all TabletGroupWatchers are started");
+      skipMigrationCheck = true;
+    } else {
+      for (TabletGroupWatcher watcher : watchers) {
+        if (!watcher.isAlive()) {
+          log.debug("Skipping migration check, not all TabletGroupWatchers are started");
+          skipMigrationCheck = true;
+          break;
+        }
```

Comment:
I had a problem with one of the ITs in this case where the TGWs wouldn't be started and it would hang on the scan. It hung every time then suddenly after a few changes it stopped hanging, so I assumed it was fixed. Looking back, I should have still been checking that the TGWs were started...

This is a good catch and looks good to me